third_party/libtock-drivers/src/nfc.rs

```diff
@@ -0,0 +1,122 @@
+use crate::result::TockResult;
+use crate::util;
+use core::cell::Cell;
+use core::mem;
+use libtock_core::{callback, syscalls};
+
+const DRIVER_NUMBER: usize = 0x30003;
+
+mod command_nr {
+    pub const TRANSMIT: usize = 1;
+    pub const RECEIVE: usize = 2;
+    pub const EMULATE: usize = 3;
+    pub const CONFIGURE: usize = 4;
+    pub const FRAMEDELAYMAX: usize = 5;
+}
+
+mod subscribe_nr {
+    pub const TRANSMIT: usize = 1;
+    pub const RECEIVE: usize = 2;
+    pub const SELECT: usize = 3;
+}
+
+mod allow_nr {
+    pub const TRANSMIT: usize = 1;
+    pub const RECEIVE: usize = 2;
+}
+
+#[allow(dead_code)]
+pub struct RecvOp {
+    pub result_code: usize,
+    pub recv_amount: usize,
+}
+
+pub struct NfcTag {}
+
+impl NfcTag {
+    pub fn enable_emulation() -> bool {
+        NfcTag::emulate(true)
+    }
+
+    pub fn disable_emulation() -> bool {
+        NfcTag::emulate(false)
+    }
+
+    fn emulate(enabled: bool) -> bool {
+        syscalls::command(DRIVER_NUMBER, command_nr::EMULATE, enabled as usize, 0).is_ok()
+    }
+
+    /// Subscribe to the tag being SELECTED callback.
+    pub fn selected() -> bool {
```

Comment:
This function will be blocking until the tag gets selected. I guess from the CTAP2 point of view, that's going to be a problematic behavior because we won't be able to receive packets from USB if we call this function.
@kaczmarczyck and @gendx  WDYT?
I have the impression that this function (as the documentation says) should only take a callback function as a parameter, issue the syscall and returns whether the subscription succeeded or not. The `yield_for` belongs to CTAP2 logic to be able to multiplex the transport channels (NFC, USB and maybe BLE at some point).
Otherwise we have to add the notion of timeouts like in USB but it means we need to poll sequentially USB and NFC which can lead to timing problems.
Last alternative would be to do the multiplexing directly in a capsule and then there's 1 single point to receive command (the capsule would probably need to tag the received packet/frame with the channel it came from for parsing/routing it properly)

Reply:
I see. They shouldn't both wait with timeout, true! Since USB was here first, I guess it has the privilege to stay blocking for now. Might make sense to revise that sometime.

Reply:
So should I change that to receive a callback and subscribe to it without yielding?

Reply:
At the moment we will move on with NFC XOR USB which removes the issue I pointed.
Multiplexing transport protocol may involve a significant amount of code changes so we'll deal with it at a later stage.

Following this decision, all NFC related code should be gated by a feature and `deploy.py` script will be responsible for ensuring that either USB or NFC are compiled but not both.